rosa/cli.py

```diff
@@ -72,11 +73,11 @@ def change_home_environment():
     os.environ["HOME"] = current_home
 
 
-def is_logged_in(aws_region=None, allowed_commands=None):
+def is_logged_in(aws_region=None, allowed_commands=None, env=None):
     _allowed_commands = allowed_commands or parse_help()
     try:
         res = build_execute_command(command="whoami", aws_region=aws_region, allowed_commands=_allowed_commands)
-        return "User is not logged in to OCM" not in res["err"]
+        return ("User is not logged in to OCM" not in res["err"]) and (res["out"]["OCM API"] == env)
```

Comment:
we need proper error for each, not logged in and logged in but to the wrong env.

Reply:
client behavior has changed; now it raises an exception.
updated the code